CP1404_practicals/prac_08/convert_miles_km.py

```diff
@@ -7,12 +7,37 @@
 from kivy.lang import Builder
 from kivy.core.window import Window
 
+MILES_TO_KM_CONVERSION = 1.60934
 
-__author__="Stuart Beattie"
 
-class MilesToKilometersApp():
+class MilesToKilometersApp(App):
     """Convert miles to kilometers GUI with plus or minus integer adjustment buttons"""
-    pass
+
+    def build(self):
+        """ build the Kivy app from the kv file """
+        Window.size = (600, 300)
+        self.title = "Convert Miles to Kilometers"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        return self.root
+
+    def handle_calculate(self, value):
+        """ handle calculation (could be button press or other call), output result to label widget """
+        result = self.convert_value(value) * MILES_TO_KM_CONVERSION
+        self.root.ids.output_label.text = "{:,.3f}".format(result)
```

Comment:
Could have used StringProperty for this part. But this wall will change the formatting, so it's understandable.